### PR TITLE
fix(tiller): adding kind to tiller client logs

### LINKED ISSUE #2162: Tiller should print out the resource kind when logging a resource name
Currently tiller only lists the resource name, but if many resources of different kinds have the same name, it becomes impossible to distinguish what tiller is referring to.

As you can see, I have some resources named with a suffix like `-secret` or `-config`, but not all, and then it becomes difficult to determine if the error is on my configmap, deployment, or something else.

Tiller Logs:
```
2017/03/22 21:13:14 release_server.go:913: Executing pre-upgrade hooks for tec-dev
2017/03/22 21:13:14 release_server.go:940: Hooks complete for pre-upgrade tec-dev
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-hookhandler-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-server-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tectonic-com-deploy-dockercfg
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-worker-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for creme-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for creme-license-key-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for dex-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-license-key-secrets
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for dex-config
2017/03/22 21:13:17 client.go:381: generating strategic merge patch for *runtime.Unstructured
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-server-config
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-worker-config
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-hookhandler-config
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for creme-config
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for dex-worker
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-nginx-redirect-domain
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for dex-overlord
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-server
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for soy-hookhandler
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for creme
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-soy-server
2017/03/22 21:13:17 client.go:381: generating strategic merge patch for *runtime.Unstructured
2017/03/22 21:13:17 client.go:246: error updating the resource tec-dev-nginx-redirect-domain:
	 delete patch type with no merge key defined
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-creme
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-dex-worker
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-soy-worker
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-dex-overlord
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-soy-hookhandler
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for tec-dev-nginx-redirect-domain-config
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for dex-worker
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for hookhandler
2017/03/22 21:13:17 client.go:393: Looks like there are no changes for creme
2017/03/22 21:13:17 release_server.go:324: warning: Upgrade "tec-dev" failed: delete patch type with no merge key defined
2017/03/22 21:13:17 storage.go:53: Updating "tec-dev" (v206) in storage
2017/03/22 21:13:17 storage.go:45: Create release "tec-dev" (v207) in storage
```

### PR DESCRIPTION
**Pull request by anubhavmishra**
Closes #2162

### CODE REVIEW

File: pkg/kube/client.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ func (c *Client) Get(namespace string, reader io.Reader) (string, error) {
 		return "", err
 	}
 	err = perform(c, namespace, infos, func(info *resource.Info) error {
-		log.Printf("Doing get for: '%s'", info.Name)
+		log.Printf("Doing get for %s: '%s'", info.Mapping.GroupVersionKind.Kind, info.Name)
```
Comment (seh) [inline pkg/kube/client.go]: - [ ] Quoting the name with single quotation marks is odd.  
  Consider instead using the `%q` format verb and removing the single quotation marks.

File: pkg/kube/client.go [hunk 2]
```diff
@@ -390,7 +390,7 @@ func updateResource(c *Client, target *resource.Info, currentObj runtime.Object,
 		return fmt.Errorf("failed to create patch: %s", err)
 	}
 	if patch == nil {
-		log.Printf("Looks like there are no changes for %s", target.Name)
+		log.Printf("Looks like there are no changes for %s %s", target.Mapping.GroupVersionKind.Kind, target.Name)
```
Comment (seh) [inline pkg/kube/client.go]: - [ ] At line 155 above, we quote the object name, but here we don't.  
  Make that treatment consistent. I recommend using the `%q` format verb for the object name.

**Review by seh — CHANGES_REQUESTED** [review]

File: pkg/kube/client.go (hunk 1 above)
Comment (anubhavmishra) [inline pkg/kube/client.go]: I didn't add the single quotation mark I just added the object type to to the print statement.

File: pkg/kube/client.go (hunk 2 above)
Comment (anubhavmishra) [inline pkg/kube/client.go]: Again. I haven't quoted the object earlier either :)

File: pkg/kube/client.go (hunk 1 above)
Comment (seh) [inline pkg/kube/client.go]: I didn't accuse _you_ of adding it. I was just suggesting what I think would be an improvement to the same line of code that you're editing here (and below at line 393). It doesn't seem worthy of a separate PR, unless you know of many more places that would require the same quoting change.

Comment (anubhavmishra) [inline pkg/kube/client.go]: will do.

**Review by thomastaylor312 — COMMENTED** [review]
I agree with @seh's feedback on this. Once that change is made, I can go ahead and merge this.

**Comment by anubhavmishra** [comment]
@thomastaylor312 done!

File: pkg/kube/client.go [hunk 3]
```diff
@@ -152,7 +152,7 @@ func (c *Client) Get(namespace string, reader io.Reader) (string, error) {
 		return "", err
 	}
 	err = perform(c, namespace, infos, func(info *resource.Info) error {
-		log.Printf("Doing get for %s: '%s'", info.Mapping.GroupVersionKind.Kind, info.Name)
+		log.Printf("Doing get for %q: %q", info.Mapping.GroupVersionKind.Kind, info.Name)
```
Comment (seh) [inline pkg/kube/client.go]: - [ ] I was only suggesting that we quote the object name, not the kind. Hence:
```Requesting %s %q via GET```

File: pkg/kube/client.go [hunk 4]
```diff
@@ -390,7 +390,7 @@ func updateResource(c *Client, target *resource.Info, currentObj runtime.Object,
 		return fmt.Errorf("failed to create patch: %s", err)
 	}
 	if patch == nil {
-		log.Printf("Looks like there are no changes for %s %s", target.Mapping.GroupVersionKind.Kind, target.Name)
+		log.Printf("Looks like there are no changes for %q %q", target.Mapping.GroupVersionKind.Kind, target.Name)
```
Comment (seh) [inline pkg/kube/client.go]: - [ ] I was only suggesting that we quote the object name, not the kind. Hence:
```Requesting %s %q via GET```

**Review by seh — CHANGES_REQUESTED** [review]
One step forward, a half step back, but that's still progress.

**Comment by seh** [comment]
I noticed these additional lines that could benefit from the same quoting treatment, in the interest of consistency in the logging style:

- line 189  
  Replace single quotation marks with bare kind name and double-quoted object name (via `%q`).
- line 236  
  Use double quotation marks around object name (via `%q`).
- line 242  
  Use double quotation marks around object name (via `%q`).
- line 246  
  Use double quotation marks around object name (via `%q`).
- line 261  
  Use double quotation marks around object name (via `%q`).
- line 263  
  Use double quotation marks around object name (via `%q`).
- line 281  
  Use double quotation marks around object name (via `%q`).
- line 353  
  Use double quotation marks around object name (via `%q`).

**Comment by anubhavmishra** [comment]
@seh I think that should look more consistent now.

**Review by thomastaylor312 — APPROVED** [review]

File: pkg/kube/client.go [hunk 5]
```diff
@@ -350,7 +350,7 @@ func deleteResource(c *Client, info *resource.Info) error {
 		}
 		return err
 	}
-	log.Printf("Using reaper for deleting %s", info.Name)
+	log.Printf("Using reaper for deleting %d", info.Name)
```
Comment (seh) [inline pkg/kube/client.go]: - [ ] s/%d/%q/

File: pkg/kube/client.go [hunk 6]
```diff
@@ -278,7 +278,7 @@ func (c *Client) Delete(namespace string, reader io.Reader) error {
 		return err
 	}
 	return perform(c, namespace, infos, func(info *resource.Info) error {
-		log.Printf("Starting delete for %s %s", info.Name, info.Mapping.GroupVersionKind.Kind)
+		log.Printf("Starting delete for %d %s", info.Name, info.Mapping.GroupVersionKind.Kind)
```
Comment (seh) [inline pkg/kube/client.go]: - [ ] s/%d/%q/

File: pkg/kube/client.go [hunk 7]
```diff
@@ -258,9 +258,9 @@ func (c *Client) Update(namespace string, originalReader, targetReader io.Reader
 	}
 
 	for _, info := range original.Difference(target) {
-		log.Printf("Deleting %s in %s...", info.Name, info.Namespace)
+		log.Printf("Deleting %d in %s...", info.Name, info.Namespace)
 		if err := deleteResource(c, info); err != nil {
-			log.Printf("Failed to delete %s, err: %s", info.Name, err)
+			log.Printf("Failed to delete %d, err: %s", info.Name, err)
```
Comment (seh) [inline pkg/kube/client.go]: - [ ] s/%d/%q/

File: pkg/kube/client.go [hunk 8]
```diff
@@ -233,17 +233,17 @@ func (c *Client) Update(namespace string, originalReader, targetReader io.Reader
...
 
 		originalInfo := original.Get(info)
 		if originalInfo == nil {
-			return fmt.Errorf("no resource with the name %s found", info.Name)
+			return fmt.Errorf("no resource with the name %d found", info.Name)
 		}
 
 		if err := updateResource(c, info, originalInfo.Object, recreate); err != nil {
-			log.Printf("error updating the resource %s:\n\t %v", info.Name, err)
+			log.Printf("error updating the resource %d:\n\t %v", info.Name, err)
```
Comment (seh) [inline pkg/kube/client.go]: - [ ] s/%d/%q/

File: pkg/kube/client.go [hunk 9]
```diff
@@ -233,17 +233,17 @@ func (c *Client) Update(namespace string, originalReader, targetReader io.Reader
...
 			kind := info.Mapping.GroupVersionKind.Kind
-			log.Printf("Created a new %s called %s\n", kind, info.Name)
+			log.Printf("Created a new %s called %d\n", kind, info.Name)
 			return nil
 		}
 
 		originalInfo := original.Get(info)
 		if originalInfo == nil {
-			return fmt.Errorf("no resource with the name %s found", info.Name)
+			return fmt.Errorf("no resource with the name %d found", info.Name)
```
Comment (seh) [inline pkg/kube/client.go]: - [ ] s/%d/%q/

File: pkg/kube/client.go [hunk 10]
```diff
@@ -233,17 +233,17 @@ func (c *Client) Update(namespace string, originalReader, targetReader io.Reader
 			}
 
 			kind := info.Mapping.GroupVersionKind.Kind
-			log.Printf("Created a new %s called %s\n", kind, info.Name)
+			log.Printf("Created a new %s called %d\n", kind, info.Name)
```
Comment (seh) [inline pkg/kube/client.go]: - [ ] s/%d/%q/

File: pkg/kube/client.go [hunk 11]
```diff
@@ -186,7 +186,7 @@ func (c *Client) Get(namespace string, reader io.Reader) (string, error) {
 		}
 		for _, o := range ot {
 			if err := p.PrintObj(o, buf); err != nil {
-				log.Printf("failed to print object type '%s', object: '%s' :\n %v", t, o, err)
+				log.Printf("failed to print object type %d, object: %d :\n %v", t, o, err)
```
Comment (seh) [inline pkg/kube/client.go]: - [ ] s/type %d, object: %d/type %s, object: %q/  
  I'm not sure what the type of variable "o" is here. If it's just a name, then the "%q" verb is appropriate. If it's a struct, then maybe "%v" would be better.

**Review by seh — COMMENTED** [review]
It looks like here you used the "%d" verb—suitable for integers—for object names. Were those typos?

**Comment by seh** [comment]
Oh, I see the next commit 76ebf20aac7b7939904130a3bf61b56e22cdf904 addresses these.

**Comment by chancez** [comment]
Woot, thanks! This will make things much nicer to debug.

**Pull request merged**